### PR TITLE
Bug 1133360 - Bring private browser ID overlay up to spec

### PR DESCRIPTION
**Pull request by mikehenrty**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[mikehenrty Michael Henretty (mikehenrty)](https://github.com/mikehenrty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=772a6ba6a698)

File: apps/system/js/app_window.js [hunk 1]
```diff
@@ -1703,8 +1703,19 @@
 
   AppWindow.prototype.getIconForSplash =
     function aw_getIconForSplash(manifest) {
-      var icons = this.manifest ?
-        ('icons' in this.manifest ? this.manifest.icons : null) : null;
+      var icons;
+      if (this.isPrivate) {
+        icons = {
+          '84': ' /style/icons/private_browser_84.png',
```
Comment (KevinGrandon) [inline apps/system/js/app_window.js]: This file does not exist yet. I've emailed Eric to ask about assets, maybe we should try to get those today/tomorrow then land this?

**Comment by try-server-hook** [comment]
[mikehenrty Michael Henretty (mikehenrty)](https://github.com/mikehenrty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4523a0ff8054)

File: apps/system/js/app_window.js [hunk 2]
```diff
@@ -1703,8 +1703,15 @@
 
   AppWindow.prototype.getIconForSplash =
     function aw_getIconForSplash(manifest) {
-      var icons = this.manifest ?
-        ('icons' in this.manifest ? this.manifest.icons : null) : null;
+      var icons;
+      if (this.isPrivate) {
```
Comment (KevinGrandon) [inline apps/system/js/app_window.js]: Hmm, not sure I like faking the icons object with just the single icon - it just feels hackish to me. What do you feel about doing an early return here instead?

```js
if (this.isPrivate) {
  this._splash = '/style/icons/pb_icon.png';
  this.preloadSplash();
  return this._splash;
}
```

File: apps/system/test/unit/app_window_test.js [hunk 3]
```diff
@@ -902,6 +902,11 @@ suite('system/AppWindow', function() {
     });
   });
 
+  test('get Private Browser Splash Icon', function() {
+    var app = new AppWindow(fakePrivateConfig);
+    assert.isTrue(app._splash.indexOf('private_browser') !== 1);
```
Comment (KevinGrandon) [inline apps/system/test/unit/app_window_test.js]: What is this meant to test? My assumption was that it was testing the old private_browser icon path which looks to be updated in this patch. I'm guessing it's catching a different string somewhere, or the URL of the splash page? Maybe add the .png to the filepath?

File: apps/system/js/app_window.js (hunk 2 above)
Comment (mikehenrty) [inline apps/system/js/app_window.js]: Yup, cleaner. I'll make that change.

File: apps/system/test/unit/app_window_test.js (hunk 3 above)
Comment (mikehenrty) [inline apps/system/test/unit/app_window_test.js]: Sloppy, yeah this needs to be updated.

**Comment by try-server-hook** [comment]
[mikehenrty Michael Henretty (mikehenrty)](https://github.com/mikehenrty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=cbca67b28c21)

File: apps/system/js/app_window.js [hunk 4]
```diff
@@ -1703,6 +1703,12 @@
 
   AppWindow.prototype.getIconForSplash =
     function aw_getIconForSplash(manifest) {
+      if (this.isPrivate) {
+        this._splash = '/style/icons/pb_icon.png';
+        this.preloadSplash();
+        return '/style/icons/pb_icon.png';
```
Comment (KevinGrandon) [inline apps/system/js/app_window.js]: nit: ```return this._splash;``` so I don't have to change this path twice in the future! :-p

I would also opt to call this private_browser_splash.png or similar so it's more readable.

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#lWIPduSySQqLDoP3jyGPxA

The pull request failed to pass integration tests. It could not be landed, please try again.

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#EAqlAVVUT_ej2Zv4kj4GQw

The pull request failed to pass integration tests. It could not be landed, please try again.

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#It3KcoG6RfyXnqvZXOMYfg

The pull request failed to pass integration tests. It could not be landed, please try again.

**Comment by KevinGrandon** [comment]
Autolander is clearly drunk. Not sure what's going on here, so let's land manually. Will investigate this.

**Pull request merged**